diff --git a/bundles/chunk-CWD5WGIF.js b/bundles/chunk-CWD5WGIF.js
index 91a9edd8e20e0b8fd2ecf626e47d28de6b88fbf7..38a66f875c151a05f1dbba09701ab05618eb3594 100755
--- a/bundles/chunk-CWD5WGIF.js
+++ b/bundles/chunk-CWD5WGIF.js
@@ -101,6 +101,7 @@ import {
   join,
   resolve
 } from "./chunk-EC5K6QPP.js";
+import { NgKitCompilerHost, NgKitProgram } from '@ngkit/compiler';
 
 // bazel-out/darwin_arm64-fastbuild/bin/packages/compiler-cli/src/transformers/api.mjs
 var DEFAULT_ERROR_CODE = 100;
@@ -7355,7 +7356,7 @@ var NgCompilerHost = class extends DelegatingCompilerHost2 {
     }
     const shimAdapter = new ShimAdapter(delegate, normalizedTsInputFiles, topLevelShimGenerators, perFileShimGenerators, oldProgram);
     const shimTagger = new ShimReferenceTagger(perFileShimGenerators.map((gen) => gen.extensionPrefix));
-    return new NgCompilerHost(delegate, inputFiles, rootDirs, shimAdapter, shimTagger, entryPoint, diagnostics);
+    return new NgKitCompilerHost(delegate, inputFiles, rootDirs, shimAdapter, shimTagger, entryPoint, diagnostics);
   }
   isShim(sf) {
     return isShim(sf);
@@ -7402,7 +7403,7 @@ var NgtscProgram = class {
       verifySupportedTypeScriptVersion();
     }
     const reuseProgram = oldProgram == null ? void 0 : oldProgram.compiler.getCurrentProgram();
-    this.host = NgCompilerHost.wrap(delegateHost, rootNames, options, reuseProgram != null ? reuseProgram : null);
+    this.host = NgKitCompilerHost.wrap(delegateHost, rootNames, options, reuseProgram != null ? reuseProgram : null);
     if (reuseProgram !== void 0) {
       retagAllTsFiles(reuseProgram);
     }
@@ -7612,7 +7613,7 @@ function mergeEmitResults(emitResults) {
 
 // bazel-out/darwin_arm64-fastbuild/bin/packages/compiler-cli/src/transformers/program.mjs
 function createProgram({ rootNames, options, host, oldProgram }) {
-  return new NgtscProgram(rootNames, options, host, oldProgram);
+  return new NgKitProgram(rootNames, options, host, oldProgram);
 }
 
 // bazel-out/darwin_arm64-fastbuild/bin/packages/compiler-cli/src/perform_compile.mjs
diff --git a/package.json b/package.json
index 1e0b67eb7f4234f7a978cec4a2bb52f36fb08205..435aa6eca9e67c54c8a7878e38441e76f864aff3 100755
--- a/package.json
+++ b/package.json
@@ -40,6 +40,10 @@
     "./private/tooling": {
       "types": "./private/tooling.d.ts",
       "default": "./bundles/private/tooling.js"
+    },
+    "./src/ngtsc/core": {
+      "types": "./src/ngtsc/core/index.d.ts",
+      "default": "./bundles/chunk-CWD5WGIF.js"
     }
   },
   "dependencies": {
