diff --git a/dist/cjs/src/cli/server-start.js b/dist/cjs/src/cli/server-start.js
index 7bd95b738f08017380f64610dcc21e70e6bfbfaf..3d2dbbe1fe1526631eceb43c142692697d9e55b4 100644
--- a/dist/cjs/src/cli/server-start.js
+++ b/dist/cjs/src/cli/server-start.js
@@ -52,7 +52,7 @@ let ServerStartController = class ServerStartController {
         const kernel = this.serviceContainer.getModule(module_js_1.FrameworkModule);
         kernel.configure(overwrite);
         const server = this.serviceContainer.getRootInjector().get(application_server_js_1.ApplicationServer);
-        await server.start(true);
+        await server.start(process.env.NODE_ENV === 'production');
     }
 };
 ServerStartController.__type = [() => __ΩLoggerInterface, 'logger', () => app_1.ServiceContainer, 'serviceContainer', 'constructor', 'host', 'port', 'workers', 'ssl', 'selfSigned', 'execute', 'Pn!2"<P7#2$<"0%P&2&8\'2\'8\'2(8)2)8)2*8$`0+5'];
diff --git a/dist/esm/src/cli/server-start.js b/dist/esm/src/cli/server-start.js
index 01cb68cd4105b9facc79c929f8f8f274179b3be1..a1e1c71d70ec561ef10c2b85f446f095149b34de 100644
--- a/dist/esm/src/cli/server-start.js
+++ b/dist/esm/src/cli/server-start.js
@@ -49,7 +49,7 @@ let ServerStartController = class ServerStartController {
         const kernel = this.serviceContainer.getModule(FrameworkModule);
         kernel.configure(overwrite);
         const server = this.serviceContainer.getRootInjector().get(ApplicationServer);
-        await server.start(true);
+        await server.start(process.env.NODE_ENV === 'production');
     }
 };
 ServerStartController.__type = [() => __ΩLoggerInterface, 'logger', () => ServiceContainer, 'serviceContainer', 'constructor', 'host', 'port', 'workers', 'ssl', 'selfSigned', 'execute', 'Pn!2"<P7#2$<"0%P&2&8\'2\'8\'2(8)2)8)2*8$`0+5'];
